src/content/popup/Words/WordTable.tsx

```diff
@@ -0,0 +1,191 @@
+import { Fragment } from 'preact';
+import { useRef } from 'preact/hooks';
+
+import type { WordResult } from '../../../background/search-result';
+
+import { NamePreview } from '../../query';
+
+import { MetadataContainer } from '../Metadata/MetadataContainer';
+import { NameEntry } from '../Names/NameEntry';
+import { CopyState } from '../copy-state';
+import { getSelectedIndex } from '../selected-index';
+import { containerHasSelectedText } from '../selection';
+import { ShowPopupOptions, StartCopyCallback } from '../show-popup';
+
+import { WordEntry } from './WordEntry';
+
+export type WordTableProps = {
+  entries: Array<WordResult>;
+  matchLen: number;
+  more: boolean;
+  namePreview?: NamePreview;
+  options: ShowPopupOptions;
+  title?: string;
+};
+
+export const WordTable = (props: WordTableProps) => {
+  const { entries, more } = props;
+
+  const wordTable = useRef<HTMLDivElement>(null);
+
+  // Pre-filter metadata
+  //
+  // If we have word matches longer than shogi metadata we drop the shogi
+  // metadata because the shogi shorthand in particular can turn up false
+  // positives on words like ドクター and ドキュメンテーション.
+  if (
+    props.options.meta?.type === 'shogi' &&
+    props.matchLen >= props.options.meta.matchLen
+  ) {
+    delete props.options.meta;
```

Comment:
I think we should probably treat `props` as read-only. So perhaps we can make up a separate `meta` local variable and mutate that? (Or simply conditionally assign it.)